### PR TITLE
Rollback transaction for any exception

### LINKED ISSUE #3832: Raiden won't start after aborting during migration
On the new RC I was trying to start Raiden, but kept on getting “Connection pool is full” warning, so I stopped Raiden (ctrl+c) during that in order to see if it would also happen with one of our own hosted nodes. These are the logs from that run of Raiden:
(https://gist.github.com/czepluch/c93314b52f22e25ddf347604247bddb3)
Then when trying to start Raiden again I am getting this error:
https://gist.github.com/czepluch/0273092ec84d28c70a32ce52b41f4d4d

Please note that this was done with an old node on mainnet that has been running older versions of Raiden previously.

Attached is the entire DB folder:
https://www.dropbox.com/s/ntuwg8cl772garu/jacob_raiden_db.zip?dl=0

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #3832 

Causes of failure:
1. `KeyboardInterrupt` was not triggering a rollback as this its parent class is a `BaseException` which wasn't included in the `except` block.
2. At some point, instantiating `SQLiteStorage` caused an implicit `update_version` in the constructor which stored `RAIDEN_DB_VERSION` in that database and overrode what the upgrade manager was expecting.

@czepluch ran the branch on his failing account and confirms the fix works.

### CODE REVIEW

**Comment by LefterisJP** [comment]
Reviewing this

File: raiden/raiden_service.py [hunk 1]
```diff
@@ -447,6 +447,7 @@ def start(self):
             database_path=self.database_path,
             serializer=serialize.JSONSerializer(),
         )
+        storage.update_version()
```
Comment (hackaugusto) [inline raiden/raiden_service.py]: This should be done by the upgrade manager, no?

File: raiden/utils/upgrades.py [hunk 2]
```diff
@@ -233,7 +233,7 @@ def _upgrade(self, target_file: str, from_file: str, from_version: int):
                         )
 
                     update_version(storage, RAIDEN_DB_VERSION)
-            except Exception as e:
+            except BaseException as e:
```
Comment (hackaugusto) [inline raiden/utils/upgrades.py]: we can use a bare-except here too

File: raiden/raiden_service.py (hunk 1 above)
Comment (rakanalh) [reply raiden/raiden_service.py]: What if there are no prior databases to migrate? i did not want to introduce a regression so we do it explicity here.

File: raiden/utils/upgrades.py (hunk 2 above)
Comment (rakanalh) [reply raiden/utils/upgrades.py]: I needed `e`

**Pull request merged**